ext/stringio/stringio.c

```diff
@@ -1583,6 +1583,48 @@ strio_read(int argc, VALUE *argv, VALUE self)
     return str;
 }
 
+/*
+ *  call-seq:
+ *    pread(maxlen, offset)             -> string
+ *    pread(maxlen, offset, out_string) -> string
+ *
+ *  See IO#pread.
+ */
+static VALUE
+strio_pread(int argc, VALUE *argv, VALUE self)
+{
+    VALUE rb_len, rb_offset, rb_buf;
+    rb_scan_args(argc, argv, "21", &rb_len, &rb_offset, &rb_buf);
+    long len = NUM2LONG(rb_len);
+    long offset = NUM2LONG(rb_offset);
+
+    if (len < 0) {
+        rb_raise(rb_eArgError, "negative string size (or size too big): %ld", len);
```

Comment:
How about using `rb_len`?

```suggestion
        rb_raise(rb_eArgError, "negative string size (or size too big): %" PRIsVALUE, rb_len);
```